File: Data Structures & Algorithms/same-binary-tree/submission-1.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:

    res = True

    def isSameTree(self, p: Optional[TreeNode], q: Optional[TreeNode]) -> bool:
        if p == None and q == None:
            return True
        if p == None and q != None or p != None and q == None or self.res == False:
            self.res = False
            return False

        
        
        if p.val != q.val or self.res == False:
            self.res = False
            return False

        check = self.isSameTree(p.left, q.left)

        check = self.isSameTree(p.right, q.right)

        return self.res

        

        

File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-1.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */

class Solution {
public:
    vector<vector<int>> levelOrder(TreeNode* root) {
        if(root == nullptr){
            return {};
        }

        vector<vector<int>> res;
        queue<TreeNode*> bfs;
        bfs.push(root);
        
        while(!empty(bfs)){
            int level = bfs.size();
            vector<int> track;

            for(int i = 0; i < level; ++i){
                TreeNode* node = bfs.front();
                bfs.pop();
                track.push_back(node->val);

                if(node->left != nullptr){
                    bfs.push(node->left);
                }
                if(node->right != nullptr){
                    bfs.push(node->right);
                }
            }

            res.push_back(track);
        }
        
        return res;
    }
};
